syntax = "proto2";

import "yandex_io/protos/model_objects.proto";
import "yandex_io/protos/remoting.proto";

option java_package = "ru.yandex.quasar.protobuf";
option java_outer_classname = "FunctionalTests";

package quasar.proto;

message EmulatedButton {
    optional EmptyMessage activate = 1;

    optional EmptyMessage volume_up = 2;
    optional EmptyMessage volume_down = 3;
    optional EmptyMessage volume_mute = 4;
    optional EmptyMessage volume_unmute = 5;
    optional EmptyMessage volume_toggle_mute = 6;
    optional EmptyMessage start_conversation = 7;
    optional EmptyMessage toggle_play_pause = 8;
    optional EmptyMessage next = 9;
    optional EmptyMessage prev = 10;
    // add other possible hardware buttons here
}

message PlatformVolumeState {
    optional int32 max_volume = 1;
    optional int32 min_volume = 2;
    optional int32 cur_volume = 3;
    optional bool muted = 4;
}

message AliceVolumeState {
    optional int32 cur_volume = 1;
    optional bool muted = 2;
}

message EmulateBluetooth {
    optional EmptyMessage connect = 1;
    optional EmptyMessage disconnect = 2;
    optional EmptyMessage play = 3;
    optional EmptyMessage pause = 4;
}

message AuthenticateRequest {
    optional string xcode = 1;
}

message TestEvent {
    enum Event {
        START = 0;
        END = 1;
        LOG = 2;
    }
    optional Event event = 1;
    optional string test_name = 2;
    optional string log = 3;
}

message TestpointMessage {
    optional EmulatedButton emulated_button = 1;
    optional EmptyMessage start_configuration_mode = 2;
    optional PlatformVolumeState platform_volume_state = 3;
    repeated Directive incoming_directive = 4;
    optional EmulateBluetooth emulated_bluetooth = 5;
    optional AuthenticateRequest authenticate_request = 6;
    optional TestEvent test_event = 7;
    optional AliceVolumeState alice_volume_state = 8;
}
